@here/harp-mapview/lib/text/TextStyleCache.ts

```diff
@@ -441,19 +459,123 @@ export class TextStyleCache {
                     style.wrappingMode === "Word"
                         ? WrappingMode[style.wrappingMode]
                         : WrappingMode.Word,
-                verticalAlignment:
-                    style.vAlignment === "Above" ||
-                    style.vAlignment === "Center" ||
-                    style.vAlignment === "Below"
-                        ? VerticalAlignment[style.vAlignment]
-                        : VerticalAlignment.Center,
-                horizontalAlignment:
-                    style.hAlignment === "Left" ||
-                    style.hAlignment === "Center" ||
-                    style.hAlignment === "Right"
-                        ? HorizontalAlignment[style.hAlignment]
-                        : HorizontalAlignment.Center
+                verticalAlignment,
+                horizontalAlignment,
+                placements
             }
         };
     }
 }
+
+function parseAlignmentAndPlacements(
+    hAlignment: string | null | undefined,
+    vAlignment: string | null | undefined,
+    placementsTokens: string | null | undefined
+): {
+    horizontalAlignment: HorizontalAlignment;
+    verticalAlignment: VerticalAlignment;
+    placements: TextPlacements;
+} {
+    // Currently supported only for PoiTechnique.
+    const placements: TextPlacements = placementsTokens
+        ? parseTechniquePlacements(placementsTokens)
+        : [];
+
+    // Ignore alignment attributes when placements attributes are defined or provide default
+    // values if none of them are defined.
+    // NOTE: Alignment ignore may be removed if we decide to support both attributes separately.
+    const horizontalAlignment =
+        placements.length > 0
+            ? hAlignFromPlacement(placements[0].h)
+            : parseTechniqueHAlignValue(hAlignment);
+
+    const verticalAlignment =
+        placements.length > 0
+            ? vAlignFromPlacement(placements[0].v)
+            : parseTechniqueVAlignValue(vAlignment);
+
+    return { horizontalAlignment, verticalAlignment, placements };
```

Comment:
Same code as in resolvePlacementAndAlignment() in TextStyle. Please refactor to have this only in one place.

Reply:
I will do, if you insist just keep in mind that with current solution it would be much faster cause the alignment attributes may be not even parsed if placement is available, moreover using the generic function (resolvePlacementAndAlignment()) implies that we need to make a new copy of TextPlacement(s) array.